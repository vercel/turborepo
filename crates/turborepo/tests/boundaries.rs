mod common;

#[test]
fn test_boundaries() -> Result<(), anyhow::Error> {
    check_json!(
        "boundaries",
        "npm@10.5.0",
        "query",
        "get boundaries lints" => "query { boundaries { items { message import } } }",
    );

    Ok(())
}
