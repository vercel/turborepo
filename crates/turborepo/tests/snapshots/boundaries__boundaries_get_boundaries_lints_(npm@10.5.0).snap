---
source: crates/turborepo/tests/boundaries.rs
expression: query_output
---
{
  "data": {
    "boundaries": {
      "items": [
        {
          "message": "cannot import file `../../packages/another/index.jsx` because it leaves the package",
          "import": "../../packages/another/index.jsx"
        },
        {
          "message": "importing from a type declaration package, but import is not declared as a type-only import",
          "import": "ship"
        },
        {
          "message": "importing from a type declaration package, but import is not declared as a type-only import",
          "import": "@types/ship"
        },
        {
          "message": "cannot import package `module-package` because it is not a dependency",
          "import": "module-package"
        }
      ]
    }
  }
}
